android-client-sdk/src/main/java/com/devcycle/sdk/android/api/Request.kt

```diff
@@ -16,12 +17,36 @@ import timber.log.Timber
 import java.io.IOException
 
 internal class Request constructor(envKey: String, apiBaseUrl: String, eventsBaseUrl: String) {
-    private val api: DVCApi = DVCApiClient().initialize(apiBaseUrl)
     private val eventApi: DVCEventsApi = DVCEventsApiClient().initialize(envKey, eventsBaseUrl)
     private val edgeDBApi: DVCEdgeDBApi = DVCEdgeDBApiClient().initialize(envKey, apiBaseUrl)
     private val objectMapper = jacksonObjectMapper()
+    private val gsonMapper = getGsonMapper()
+    private val api: DVCApi = DVCApiClient().initialize(apiBaseUrl, gsonMapper)
     private val configMutex = Mutex()
 
+    private fun getGsonMapper(): Gson {
+        return GsonBuilder().registerTypeAdapter(Variable.TypeEnum::class.java, object : TypeAdapter<Variable.TypeEnum>() {
```

Comment:
is all this necessary to make the enums work?

Reply:
let me double check, we need this type adapter change for sure.. are we missing any other TypeEnums? 🤔 

- but I don't believe we need the `@SerializedNames("0")`, going to verify that

Reply:
any idea why we need it? does gson not auto-infer stuff from the enum types?

Reply:
yeah, it doesn't auto-infer from enums 😢 our unit tests caught this

Reply:
Specifically enums are an issue yes.